system/system_monitor/src/hdd_monitor/hdd_monitor.cpp

```diff
@@ -477,5 +690,173 @@ void HDDMonitor::updateHDDInfoList()
+}
+
+double HDDMonitor::getIncreaseSysfsDeviceStatValuePerSec(
+  uint64_t cur_val, uint64_t last_val, double duration_sec)
+{
+  if (cur_val > last_val && duration_sec > 0.0) {
+    return static_cast<double>(cur_val - last_val) / duration_sec;
+  }
+  return 0.0;
+}
+
+int HDDMonitor::readSysfsDeviceStat(const std::string & device, SysfsDevStat & sfdevstat)
+{
+  int ret = -1;
+  unsigned int ios_pgr, tot_ticks, rq_ticks, wr_ticks;
+  uint64_t rd_ios, rd_merges_or_rd_sec, wr_ios, wr_merges;
+  uint64_t rd_sec_or_wr_ios, wr_sec, rd_ticks_or_wr_sec;
+  uint64_t dc_ios, dc_merges, dc_sec, dc_ticks;
+
+  std::string filename("/sys/block/");
+  filename += device + "/stat";
+  FILE * fp = fopen(filename.c_str(), "r");
+  if (fp == NULL) {
+    return ret;
+  }
+
+  int i = fscanf(
+    fp, "%lu %lu %lu %lu %lu %lu %lu %u %u %u %u %lu %lu %lu %lu", &rd_ios, &rd_merges_or_rd_sec,
+    &rd_sec_or_wr_ios, &rd_ticks_or_wr_sec, &wr_ios, &wr_merges, &wr_sec, &wr_ticks, &ios_pgr,
+    &tot_ticks, &rq_ticks, &dc_ios, &dc_merges, &dc_sec, &dc_ticks);
+
+  if (i >= 7) {
+    sfdevstat.rd_ios_ = rd_ios;
+    sfdevstat.rd_sectors_ = rd_sec_or_wr_ios;
+    sfdevstat.wr_ios_ = wr_ios;
+    sfdevstat.wr_sectors_ = wr_sec;
+    ret = 0;
+  }
+
+  fclose(fp);
+  return ret;
+}
+
+void HDDMonitor::updateHDDConnections()
+{
+  for (auto & hdd_param : hdd_params_) {
+    // Get device name from mount point
+    hdd_param.second.part_device_ = getDeviceFromMountPoint(hdd_param.first);
+    if (!hdd_param.second.part_device_.empty()) {
+      // Check the existence of device file
+      std::error_code ec;
+      if (std::filesystem::exists(hdd_param.second.part_device_, ec)) {
+        hdd_connected_flags_[hdd_param.first] = true;
+
+        // Remove index number of partition for passing device name to hdd-reader
+        if (boost::starts_with(hdd_param.second.part_device_, "/dev/sd")) {
+          const std::regex pattern("\\d+$");
+          hdd_param.second.disk_device_ =
+            std::regex_replace(hdd_param.second.part_device_, pattern, "");
+        } else if (boost::starts_with(hdd_param.second.part_device_, "/dev/nvme")) {
+          const std::regex pattern("p\\d+$");
+          hdd_param.second.disk_device_ =
+            std::regex_replace(hdd_param.second.part_device_, pattern, "");
+        }
+
+        const std::regex raw_pattern(".*/");
+        hdd_stats_[hdd_param.first].device_ =
+          std::regex_replace(hdd_param.second.disk_device_, raw_pattern, "");
+
+        continue;
+      } else {
+        // Unmount to solve the state where the device is mounted without existing
+        if (unmountDeviceWithLazy(hdd_param.second.part_device_)) {
+          RCLCPP_ERROR(
+            get_logger(), "Failed to unmount. %s", hdd_param.second.part_device_.c_str());
+        }
+      }
+    }
+    hdd_connected_flags_[hdd_param.first] = false;
+  }
+}
+
+int HDDMonitor::unmountDeviceWithLazy(std::string & device)
```

Comment:
@v-nakayama7440-esol 
This will be failed with unprivileged user. Please think of another way.

Reply:
@ito-san 
I moved unmount function to hdd_reader.
b679b8a57aff29f39d265a14a8892382e4d0bc14

Reply:
Verified the issue was gone after the fix.